### ISSUE TITLE
Training loss not dropping after update code with yolov3

### ISSUE THREAD

**Issue by colinlin1982** [AlexeyAB/darknet#4455]
Thank you for this excellent work.
I update code from 2019/08/09 to 2019/12/04, and continue training with existed yolov3 model on my dataset.
My config file like this:
`[net]
# Testing
#batch=1
#subdivisions=1
# Training
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 520200
policy=steps
steps=450000,500000
scales=.1,.1

...

[yolo]
mask = 6,7,8
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=1
num=9
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
`
And I found the loss scaled down immediately and not dropping over 20000 bathes.
My chart.png(max loss=0.1):
![chart](https://user-images.githubusercontent.com/13325086/70288500-d62a6c00-180c-11ea-81f2-64766e1e1f32.png)

My question is:
1. Why the loss immediately scale down, have you changed the MSE loss function?
2. Does the loss look normal? Shall I change the learning rate or replace MSE with GIOU loss?

**Comment by aql315** [comment]
try with subdivisions=2

**Comment by AlexeyAB** [comment]
There was a minor fix in loss, now ignore_thresh uses `best_match_iou` instead of `best_iou`, but it should increase both Loss and Accuracy https://github.com/AlexeyAB/darknet/blob/ebf1d7a73b0d996378de17606425d77a6cea166b/src/yolo_layer.c#L353-L355

Also there was solved a bug of forced blurring, so may be this is the reason.

**Comment by colinlin1982** [comment]
> There was a minor fix in loss, now ignore_thresh uses `best_match_iou` instead of `best_iou`, but it should increase both Loss and Accuracy
> 
> https://github.com/AlexeyAB/darknet/blob/ebf1d7a73b0d996378de17606425d77a6cea166b/src/yolo_layer.c#L353-L355
> 
> Also there was solved a bug of forced blurring, so may be this is the reason.

I add blur=1 in net config, it does not change much.
I have change yolo loss to GIOU, it works fine.
Also I have try Gaussian Yolo layer, but run into nan a few batches. I think there must be something wrong with Gaussian Yolo layer with empty sample images.

**Comment by AlexeyAB** [comment]
> Also I have try Gaussian Yolo layer, but run into nan a few batches. 

Decrease learning rate x10.

> . I think there must be something wrong with Gaussian Yolo layer with empty sample images.

Gaussian Yolo part isn't applied for empty images, just will be applied delta for objectness.

**Comment by colinlin1982** [comment]
> > Also I have try Gaussian Yolo layer, but run into nan a few batches.
> 
> Decrease learning rate x10.
Decrease learning rate from 0.00261 to 0.000261, still run into NAN around 64000 images.
> 
> > . I think there must be something wrong with Gaussian Yolo layer with empty sample images.
> 
> Gaussian Yolo part isn't applied for empty images, just will be applied delta for objectness.
I have removed objectness images(not containing classified object and .txt file is empty) from train.txt, and it works fine. 
[net]
# Testing
#batch=1
#subdivisions=1
# Training
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.949
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1
blur=1

learning_rate=0.000261
burn_in=1000
max_batches = 520200
policy=steps
steps=450000,500000
scales=.1,.1
max_epochs = 300

...
[Gaussian_yolo]
mask = 6,7,8
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=1
num=9
jitter=.3
ignore_thresh=.5
truth_thresh=1
iou_thresh=0.213
uc_normalizer=1.0
cls_normalizer=1.0
iou_normalizer=0.5
iou_loss=giou
scale_x_y=1.0
random=1
